src/executor.rs

```diff
@@ -1,8 +1,6 @@
 use std::future::Future;
 
-use crate::fs::FileProvider;
-
-pub trait Executor: FileProvider {
+pub trait Executor {
```

Comment:
Executor has to provide the local file system implementation, use tokio executor + monoio filesystem should cause panic.